### ISSUE TITLE
How to filter only comments that are not in an thread

### ISSUE THREAD

**Issue by OliveiraCleidson** [VirtusLab-Open-Source/strapi-plugin-comments#201]
# How to filter only comments that are not in a thread

I'm working on one feature that gets comments with pagination. I won't get comments sorted by created date and only comments that are root comments, then I'm using the REST API in flat mode. 

## Bugs (?)

But when I try to filter using `threadOf` the server returns 5xx.
![image](https://user-images.githubusercontent.com/40809069/230492437-351edfe8-2597-4996-b52f-d72d74157f55.png)

When I use `threadOf` with id, the `threadOf` field does not return the correct content inner `threadOf`, but the filter works.
![image](https://user-images.githubusercontent.com/40809069/230493668-97070858-cec8-4ea0-b708-069be634a393.png)

When I try to use `threadFirstItemId`
![image](https://user-images.githubusercontent.com/40809069/230493925-e0d1f808-975c-4342-b988-3dae22fe819c.png)

![image](https://user-images.githubusercontent.com/40809069/230493987-ac8a9456-bc60-4a26-9bc2-4d3816423b61.png)

My dependencies:
```json
{
"dependencies": {
      "@strapi-community/strapi-provider-upload-google-cloud-storage": "^4.3.6",
      "@strapi/plugin-documentation": "^4.4.1",
      "@strapi/plugin-i18n": "4.6.1",
      "@strapi/plugin-sentry": "^4.5.0",
      "@strapi/plugin-users-permissions": "4.6.1",
      "@strapi/strapi": "4.6.1",
      "elastic-apm-node": "^3.40.1",
      "pg": "8.6.0",
      "sharp": "^0.31.2",
      "strapi-plugin-comments": "^2.2.4",
      "strapi-plugin-import-export-entries": "^1.18.0"
  }
}
  ```

My `plugin.ts`
```typescript
comments: {
    enabled: true,
    config: {
      // ##Documentation https://market.strapi.io/plugins/strapi-plugin-comments
    },
}
```

My node version Is *16.17.0*

**Comment by cyp3rius** [comment]
Hello @OliveiraCleidson ,

So step by step :)

1. `filters[threadOf][$null]=true`

It causes 500 because of following error and that's purely Strapi error. I assume the case here is with relation property.

```{
  threadOf: { '$null': 'true' },
  '$or': [ { removed: [Object] }, { removed: false } ],
  related: 'api::page.page:1'
}
[2023-04-07 15:47:25.215] error: Only $and, $or and $not can only be used as root level operators. Found $null.
Error: Only $and, $or and $not can only be used as root level operators. Found $null.
    at processWhere (/Users/mziarko/Projects/open-source/strapi-v4/node_modules/@strapi/database/lib/query/helpers/where.js:140:13)
    at processNested (/Users/mziarko/Projects/open-source/strapi-v4/node_modules/@strapi/database/lib/query/helpers/where.js:116:12)
    at processWhere (/Users/mziarko/Projects/open-source/strapi-v4/node_modules/@strapi/database/lib/query/helpers/where.js:161:25)```
    
2. `filters[threadOf][id][$null]=true`

This returned value is our error in reality as we're getting 1:1 from query. I will fix that in next release, but such query is correct.

3. The `threadFirstItemId` is not DB field but calculated property. You can't query by it.

**Comment by cyp3rius** [comment]
Nope, there is no such dedicated method - ideally it should be done by the way you described.

If you can fix point 2. we would be grateful :)

**Comment by cyp3rius** [comment]
Released as part of `v2.2.10`
